### PR TITLE
CXX-489 add inline macros

### PR DESCRIPTION
**Pull request by amidvidy**

### CODE REVIEW

**Comment by amidvidy** [comment]
note that we can't include the inline keyword because we also need this when functions are defined at their declarations. libc++ does the same thing.

**Comment by acmorrow** [comment]
LGTM

File: src/bsoncxx/builder/basic/impl.hpp [hunk 1]
```diff
@@ -57,6 +60,7 @@ generic_append(core* core, T&& t) {
 }  // namespace
 
 template <typename T>
+BSONCXX_INLINE
```
Comment (TylerBrock) [inline src/bsoncxx/builder/basic/impl.hpp]: Might be nicer on the same line as template keyword?

Comment (amidvidy) [inline src/bsoncxx/builder/basic/impl.hpp]: i kind of like having it on its own line for templates so it doesn't distract from the template types.

**Pull request merged**